### ISSUE TITLE
Trouble submitting port request from Ops

### ISSUE THREAD

**Issue by danry25** [AccelerateNetworks/NumberSearch#253]
**Describe the bug**
Trying to submit a 21 phone number port request fails due to a BulkVS API call response we can't process. No port request (aka LNP order) is created with BulkVS.

**To Reproduce**
1. Go to Quotes
2. Click on View Port Request 
3. Scroll down to Submit to Teli and BulkVS
4. See no result for port submission

**Expected behavior**
The port should be successfully submitted to the carrier.

**Error log**
```
2021-12-22 15:36:52.691 -08:00 [INF] Request starting HTTP/1.1 GET http://ops.acceleratenetworks.com/Home/PortRequests/1c9e058b-9700-4803-9bc3-8df3e662a896 - -
2021-12-22 15:36:52.693 -08:00 [INF] Executing endpoint 'NumberSearch.Ops.Controllers.HomeController.PortRequests (NumberSearch.Ops)'
2021-12-22 15:36:52.693 -08:00 [INF] Route matched with {action = "PortRequests", controller = "Home", page = "", area = ""}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.A$
2021-12-22 15:36:52.700 -08:00 [INF] Executing ViewResult, running view PortRequestEdit.
2021-12-22 15:36:52.708 -08:00 [INF] Executed ViewResult - view PortRequestEdit executed in 7.334ms.
2021-12-22 15:36:52.708 -08:00 [INF] Executed action NumberSearch.Ops.Controllers.HomeController.PortRequests (NumberSearch.Ops) in 14.6288ms
2021-12-22 15:36:52.709 -08:00 [INF] Executed endpoint 'NumberSearch.Ops.Controllers.HomeController.PortRequests (NumberSearch.Ops)'
2021-12-22 15:36:52.709 -08:00 [INF] Request finished HTTP/1.1 GET http://ops.acceleratenetworks.com/Home/PortRequests/1c9e058b-9700-4803-9bc3-8df3e662a896 - - - 200 - text/html;+charset=utf-8 18.2084ms
2021-12-22 15:36:53.039 -08:00 [INF] Request starting HTTP/1.1 GET http://ops.acceleratenetworks.com/css/site.css - -
2021-12-22 15:36:53.040 -08:00 [INF] Sending file. Request path: '/css/site.css'. Physical path: '/var/www/NumberSearch.Ops/wwwroot/css/site.css'
2021-12-22 15:36:53.042 -08:00 [INF] Request finished HTTP/1.1 GET http://ops.acceleratenetworks.com/css/site.css - - - 200 1779 text/css 3.6300ms
2021-12-22 15:36:55.637 -08:00 [INF] Request starting HTTP/1.1 POST http://ops.acceleratenetworks.com/Home/PortRequestUnified/1c9e058b-9700-4803-9bc3-8df3e662a896 application/x-www-form-urlencoded 270
2021-12-22 15:36:55.638 -08:00 [INF] Executing endpoint 'NumberSearch.Ops.Controllers.HomeController.UnifiedPortRequestAsync (NumberSearch.Ops)'
2021-12-22 15:36:55.639 -08:00 [INF] Route matched with {action = "UnifiedPortRequest", controller = "Home", page = "", area = ""}. Executing controller action with signature System.Threading.Tasks.Task`1[Micro$
2021-12-22 15:36:58.101 -08:00 [ERR] [PortRequest] Failed to submit port request to BulkVS.
2021-12-22 15:36:58.102 -08:00 [ERR] Response could not be deserialized to JSON: PUT https://portal.bulkvs.com/api/v1.0/portTn
2021-12-22 15:36:58.159 -08:00 [ERR]    at Flurl.Http.FlurlRequest.HandleExceptionAsync(FlurlCall call, Exception ex, CancellationToken token)
   at Flurl.Http.FlurlResponse.GetJsonAsync[T]()
   at NumberSearch.DataAccess.BulkVS.PortTnRequest.PutAsync(String username, String password) in /home/vsts/work/1/s/NumberSearch.DataAccess/BulkVS/PortTn.cs:line 189
   at NumberSearch.Ops.Controllers.HomeController.UnifiedPortRequestAsync(String OrderId) in /home/vsts/work/1/s/NumberSearch.Ops/Controllers/HomeController.cs:line 3075
2021-12-22 15:36:58.884 -08:00 [ERR] [PortRequest] Failed to submit port request to BulkVS.
2021-12-22 15:36:58.885 -08:00 [ERR] Response could not be deserialized to JSON: PUT https://portal.bulkvs.com/api/v1.0/portTn
2021-12-22 15:36:58.885 -08:00 [ERR]    at Flurl.Http.FlurlRequest.HandleExceptionAsync(FlurlCall call, Exception ex, CancellationToken token)
   at Flurl.Http.FlurlResponse.GetJsonAsync[T]()
   at NumberSearch.DataAccess.BulkVS.PortTnRequest.PutAsync(String username, String password) in /home/vsts/work/1/s/NumberSearch.DataAccess/BulkVS/PortTn.cs:line 189
   at NumberSearch.Ops.Controllers.HomeController.UnifiedPortRequestAsync(String OrderId) in /home/vsts/work/1/s/NumberSearch.Ops/Controllers/HomeController.cs:line 3075
2021-12-22 15:36:58.889 -08:00 [INF] [Port Request] Updated Order 1c9e058b-9700-4803-9bc3-8df3e662a896 to kick off the background work.
```

**Comment by uncheckederror** [comment]
{StatusCode: 409, ReasonPhrase: 'Conflict', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  Server: nginx/1.18.0
  Date: Tue, 28 Dec 2021 00:28:08 GMT
  Connection: keep-alive
  Content-Type: application/json; charset=utf-8
  Content-Length: 656
}}

The SPID for 18 of the numbers in this order comes back as null when we do a lookup on them. This is the response from BulkVS's porting endpoint when we submit the numbers with a null SPID as a group for porting.

**Comment by uncheckederror** [comment]
"\n{\n    \"TN Groups\": {\n        \"IP HORIZON LLC\": [\n            \"12066578423\",\n            \"12068448444\",\n            \"15033436135\",\n            \"15645297786\",\n            \"12068447605\",\n            \"15645297783\",\n            \"12069845588\",\n            \"15645297773\",\n            \"15645297874\",\n            \"15645297823\",\n            \"15645297799\",\n            \"12064240707\",\n            \"14259797549\",\n            \"14259797518\",\n            \"12066724477\",\n            \"15645297758\",\n            \"14259797539\"\n        ],\n        \"\": [\n            \"12068488838\"\n        ]\n    },\n    \"Code\": \"7401\",\n    \"Description\": \"Numbers Cannot be ported on one Order\"\n}"

**Comment by uncheckederror** [comment]
![image](https://user-images.githubusercontent.com/11726956/147516399-83941ae8-bbbb-4081-b58a-9c65112f62ee.png)

One of the numbers appears to be invalid. We only group numbers for port requests based on the SPID so if no SPID is returned by the lookup they all get grouped together.

**Comment by danry25** [comment]
Confirmed vendor issue, the vendor has resolved this issue: [ { "DialedNumber": "2068482888", "OldSPID": "", "CurrentSPID": "505B", "OldSPIDName": "", "CurrentSPIDName": "ONVOY, LLC - WA" } ]
